### PR TITLE
Disable LocalHistoryZeroSuggestProvider when history suggestions are disabled

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/26894

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Do a search for `clouds`
2. Click the omnibox on the NTP with no text entered. A search suggestion for `clouds` should show up
3. In chrome://settings/appearance disable history suggestions
4. On the NTP, click the omnibox with no text entered. The search suggestion for `clouds` should **NOT** show up.

### CODE REVIEW

File: components/omnibox/browser/brave_local_history_zero_suggest_provider.cc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/omnibox/browser/brave_local_history_zero_suggest_provider.h"
+
+#include "brave/components/omnibox/browser/brave_omnibox_prefs.h"
+#include "components/omnibox/browser/autocomplete_input.h"
+#include "components/omnibox/browser/autocomplete_provider_client.h"
+#include "components/omnibox/browser/local_history_zero_suggest_provider.h"
+#include "components/prefs/pref_service.h"
```
Comment (simonhong) [inline components/omnibox/browser/brave_local_history_zero_suggest_provider.cc:12]: need to add `//components/prefs` to `brave_components_omnibox_browser_deps`

**Review by simonhong — APPROVED** [review]
++

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: components/omnibox/browser/brave_local_history_zero_suggest_provider.cc [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    AutocompleteProviderListener* listener) {
+  return new BraveLocalHistoryZeroSuggestProvider(client, listener);
+}
+
+BraveLocalHistoryZeroSuggestProvider::~BraveLocalHistoryZeroSuggestProvider() =
+    default;
+
+void BraveLocalHistoryZeroSuggestProvider::Start(const AutocompleteInput& input,
+                                                 bool minimal_changes) {
+  if (!client_->GetPrefs()->GetBoolean(omnibox::kHistorySuggestionsEnabled)) {
```
Comment (goodov) [inline components/omnibox/browser/brave_local_history_zero_suggest_provider.cc:27]: a test would be nice for this.

Comment (fallaciousreasoning) [reply components/omnibox/browser/brave_local_history_zero_suggest_provider.cc:27]: Added `brave_local_history_zero_suggest_provider_unittest.cc`

**Review by goodov — DISMISSED** [review]

**Comment by fallaciousreasoning** [comment]
Whoops sorry @goodov, could you take another look?

File: components/omnibox/browser/brave_local_history_zero_suggest_provider_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+#include "components/omnibox/browser/in_memory_url_index.h"
+#include "components/omnibox/browser/test_scheme_classifier.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/testing_pref_service.h"
+#include "components/search_engines/prepopulated_engines.h"
+#include "components/search_engines/template_url.h"
+#include "components/search_engines/template_url_data_util.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/googletest/src/googletest/include/gtest/gtest.h"
```
Comment (goodov) [inline components/omnibox/browser/brave_local_history_zero_suggest_provider_unittest.cc:34]: `testing/gtest/include/gtest/gtest.h` is correct. this one is not needed.

**Review by goodov — APPROVED** [review]

**Pull request merged**